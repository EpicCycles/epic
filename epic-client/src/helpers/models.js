export const BRAND = "brand";
export const FRAME_NAME = "frame_name";
export const MODEL_NAME = "model_name";
export const DESCRIPTION = "description";
export const COLOURS = "colours";
export const SELL_PRICE = "sell_price";
export const SIZES = "sizes";

export const BRAND_FIELD = { fieldName: BRAND, header: "Brand", synonyms: [BRAND, 'manufacturer'] };
export const FRAME_NAME_FIELD = { fieldName: FRAME_NAME, header: "Frame Name" };
export const MODEL_NAME_FIELD = { fieldName: MODEL_NAME, header: "Model Name" };
export const COLOURS_FIELD = {
    fieldName: COLOURS,
    header: "Colours",
    synonyms: [COLOURS, "colour", "colors", "color"]
};
export const DESCRIPTION_FIELD = { fieldName: DESCRIPTION, header: "Description", synonyms: [DESCRIPTION, "desc"] };
export const SELL_PRICE_FIELD = {
    fieldName: SELL_PRICE,
    header: "Selling Price",
    synonyms: ["selling price", "srp", "rrp", "sell price", "retail price"]
};
export const SIZES_FIELD = {
    fieldName: SIZES,
    header: "Sizes",
    synonyms: [SIZES, "size", "frame sizes", "frame size"]
};
export const frameFields = [BRAND, FRAME_NAME];
export const bikeFields = [MODEL_NAME_FIELD, DESCRIPTION_FIELD, COLOURS_FIELD, SELL_PRICE_FIELD, SIZES_FIELD];