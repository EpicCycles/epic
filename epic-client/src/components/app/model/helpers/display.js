export const fixedHeaderClassname = (lockColumn) => {
    if (lockColumn) return "grid-header--fixed-left";
    return "";
};
export const fixedDetailsClassname = (lockColumn) => {
    if (lockColumn) return "grid-item--fixed-left";
    return "";
};
export const gridHeaderClass = (baseClassName = "", fieldIndex, firstColumnLocked) => {
  const shouldLock = firstColumnLocked && (fieldIndex === 0);
    return `${baseClassName} grid-item--header ${fixedHeaderClassname(shouldLock)}`;
};
export const gridItemClass = (baseClassName = "", fieldIndex, firstColumnLocked) => {
  const shouldLock = firstColumnLocked && (fieldIndex === 0);
    return `${baseClassName} grid-item ${fixedDetailsClassname(shouldLock)}`;
};