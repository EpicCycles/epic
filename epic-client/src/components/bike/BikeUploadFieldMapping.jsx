import React from "react";

/**
 *
 * @type {Function}
 * props
 * field, index, allowDrop, assignToBikeAttribute, rowMappings,undoMapping
 */
import {BikeUploadMatchedData} from "./BikeUploadMatchedData";

export const BikeUploadFieldMapping = (props) => {
    return (<div className="grid-row">
        <div
            className="grid-item--borderless field-label align_right"
            key={`bikeField${props.index}`}
            onDragOver={event => props.allowDrop(event)}
            onDrop={event => props.assignToBikeAttribute(event, props.field.fieldName)}
        >
            {props.field.header}
        </div>
        <div
            key={`matchFieldDiv${props.index}`}
            className="grid-item--borderless field-label "
        >
            {props.rowMappings
                .map((matched, matchIndex) => <BikeUploadMatchedData
                    key={`matchArea${props.index}${matchIndex}`}
                    matched={matched}
                    matchIndex={matchIndex}
                    undoMapping={props.undoMapping}
                />)
            }
        </div>
    </div>);
};