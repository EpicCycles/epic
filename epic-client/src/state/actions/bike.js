export const CLEAR_FRAME = "bike/CLEAR_FRAME";
export const BIKE_REVIEW_REQUESTED = "bike/BIKE_REVIEW_REQUESTED";
export const BIKE_REVIEW_BIKE = "bike/BIKE_REVIEW_BIKE";
export const BIKE_DELETE_REQUESTED = "bike/BIKE_DELETE_REQUESTED";
export const BIKE_DELETE_OK = "bike/BIKE_DELETE_OK";
export const BIKE_DELETE_ERROR = "bike/BIKE_DELETE_ERROR";
export const BIKE_ADD_PART_REQUESTED = "bike/BIKE_ADD_PART_REQUESTED";
export const BIKE_ADD_PART_OK = "bike/BIKE_ADD_PART_OK";
export const BIKE_ADD_PART_ERROR = "bike/BIKE_ADD_PART_ERROR";
export const BIKE_SAVE_REQUESTED = "bike/BIKE_SAVE_REQUESTED";
export const BIKE_SAVE_OK = "bike/BIKE_SAVE_OK";
export const BIKE_SAVE_ERROR = "bike/BIKE_SAVE_ERROR";
export const BIKE_PART_DELETE_REQUESTED = "bike/BIKE_PART_DELETE_REQUESTED";
export const BIKE_PART_DELETE_OK = "bike/BIKE_PART_DELETE_OK";
export const BIKE_PART_DELETE_ERROR = "bike/BIKE_PART_DELETE_ERROR";
export const BIKE_PART_SAVE_REQUESTED = "bike/BIKE_PART_SAVE_REQUESTED";
export const BIKE_PART_SAVE_OK = "bike/BIKE_PART_SAVE_OK";
export const BIKE_PART_SAVE_ERROR = "bike/BIKE_PART_SAVE_ERROR";
export const FRAME_ARCHIVE_REQUESTED = "bike/FRAME_ARCHIVE_REQUESTED";
export const FRAME_ARCHIVE_OK = "bike/FRAME_ARCHIVE_OK";
export const FRAME_ARCHIVE_ERROR = "bike/FRAME_ARCHIVE_ERROR";
export const FRAME_DELETE_REQUESTED = "bike/FRAME_DELETE_REQUESTED";
export const FRAME_DELETE_OK = "bike/FRAME_DELETE_OK";
export const FRAME_DELETE_ERROR = "bike/FRAME_DELETE_ERROR";
export const FRAME_SAVE_REQUESTED = "bike/FRAME_SAVE_REQUESTED";
export const FRAME_SAVE_OK = "bike/FRAME_SAVE_OK";
export const FRAME_SAVE_ERROR = "bike/FRAME_SAVE_ERROR";
export const FRAME_UPLOAD_REQUESTED = "bike/FRAME_UPLOAD_REQUESTED";
export const FRAME_UPLOAD_OK = "bike/FRAME_UPLOAD_OK";
export const FRAME_UPLOAD_ERROR = "bike/FRAME_UPLOAD_ERROR";
export const FRAME_LIST_REQUESTED = "bike/FRAME_LIST_REQUESTED";
export const FRAME_LIST_OK = "bike/FRAME_LIST_OK";
export const FRAME_LIST_ERROR = "bike/FRAME_LIST_ERROR";


export const clearFrame = () => ({
    type: CLEAR_FRAME
});
export const saveBike = (bike) => ({
    type: BIKE_SAVE_REQUESTED,
    payload: { bike }
});
export const saveBikeOK = (bike) => ({
    type: BIKE_SAVE_OK,
    payload: { bike }
});
export const saveBikeError = (error) => ({
    type: BIKE_SAVE_ERROR,
    payload: error
});
export const saveBikePart = (bikeId, part) => ({
    type: BIKE_PART_SAVE_REQUESTED,
    payload: { bikeId, part }
});
export const saveBikePartOK = (parts) => ({
    type: BIKE_PART_SAVE_OK,
    payload: { parts }
});
export const saveBikePartError = (error) => ({
    type: BIKE_PART_SAVE_ERROR,
    payload: error
});
export const addBikePart = (bikeId, part) => ({
    type: BIKE_ADD_PART_REQUESTED,
    payload: { bikeId, part }
});
export const addBikePartOK = (parts) => ({
    type: BIKE_ADD_PART_OK,
    payload: { parts }
});
export const addBikePartError = (error) => ({
    type: BIKE_ADD_PART_ERROR,
    payload: error
});
export const deleteBikePart = (bikeId, partId) => ({
    type: BIKE_PART_DELETE_REQUESTED,
    payload: { bikeId, partId }
});
export const deleteBikePartOK = (parts) => ({
    type: BIKE_PART_DELETE_OK,
    payload: { parts }
});
export const deleteBikePartError = (error) => ({
    type: BIKE_PART_DELETE_ERROR,
    payload: error
});
export const reviewBikes = (bikeReviewList) => ({
    type:BIKE_REVIEW_REQUESTED,
    payload: {bikeReviewList}
});
export const reviewBike = (bikeId) => ({
    type:BIKE_REVIEW_BIKE,
    payload: { bikeId }
});

export const deleteBikes = (bikeDeleteList, searchCriteria) => ({
    type: BIKE_DELETE_REQUESTED,
    payload: { bikeDeleteList, searchCriteria }
});
export const deleteBikesError = (error) => ({
    type: BIKE_DELETE_ERROR,
    payload: error
});
export const deleteBikesSuccess = () => ({
    type: BIKE_DELETE_OK
});
export const archiveFrames = (frameArchiveList, searchCriteria) => ({
    type: FRAME_ARCHIVE_REQUESTED,
    payload: { frameArchiveList, searchCriteria }
});
export const archiveFramesError = (error) => ({
    type: FRAME_ARCHIVE_ERROR,
    payload: error
});
export const archiveFramesSuccess = () => ({
    type: FRAME_ARCHIVE_OK
});
export const deleteFrames = (frameDeleteList, searchCriteria) => ({
    type: FRAME_DELETE_REQUESTED,
    payload: { frameDeleteList, searchCriteria }
});
export const deleteFramesError = (error) => ({
    type: FRAME_DELETE_ERROR,
    payload: error
});
export const deleteFramesSuccess = () => ({
    type: FRAME_DELETE_OK
});
export const saveFrame = (frame, searchCriteria) => ({
    type: FRAME_SAVE_REQUESTED,
    payload: { frame, searchCriteria }
});
export const saveFrameError = (error) => ({
    type: FRAME_SAVE_ERROR,
    payload: error
});
export const saveFrameSuccess = (frame) => ({
    type: FRAME_SAVE_OK,
    payload: frame
});
export const uploadFrame = (frame) => ({
    type: FRAME_UPLOAD_REQUESTED,
    payload: { frame }
});
export const uploadFrameError = (error) => ({
    type: FRAME_UPLOAD_ERROR,
    payload: error
});
export const uploadFrameSuccess = () => ({
    type: FRAME_UPLOAD_OK,
});
export const getFrameList = (searchCriteria) => ({
    type: FRAME_LIST_REQUESTED,
    payload: searchCriteria
});
export const getFrameListOK = (apiData) => ({
    type: FRAME_LIST_OK,
    payload: apiData
});
export const getFrameListError = (error) => ({
    type: FRAME_LIST_ERROR,
    payload: error
});
