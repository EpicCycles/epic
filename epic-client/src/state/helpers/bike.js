export const replaceBikeParts = (bikeId, newBikeParts, existingBikeParts) => {
    let bikePartsForState = existingBikeParts.filter(oldBikePart => oldBikePart.bike !== bikeId);
    return bikePartsForState.concat(newBikeParts);
};