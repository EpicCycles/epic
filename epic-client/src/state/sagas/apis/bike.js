import api from '../api';

const saveFrame = async (payload) => {
    api.instance.defaults.headers.common['Authorization'] = `Token ${payload.token}`;
    return await api.instance.patch(`api/frames/${payload.frame.id}`, payload.frame);
};

const uploadFrame = async (payload) => {
    api.instance.defaults.headers.common['Authorization'] = `Token ${payload.token}`;
    return await api.instance.post("api/frame/upload/", payload.frame);
};

const getFrames = async (payload) => {
    api.instance.defaults.headers.common['Authorization'] = `Token ${payload.token}`;
    const frameApi = `/api/frames?brand=${payload.brand}&frameName=${payload.frameName}&archived=${payload.archived}`;
    return await api.instance.get(frameApi);
};

export default {
    uploadFrame,
    getFrames,
    saveFrame,
}
