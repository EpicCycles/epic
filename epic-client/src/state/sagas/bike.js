import {call, put, select, takeLatest} from "redux-saga/effects";
import * as selectors from "../selectors/user";
import bike from "./apis/bike";
import history from "../../history";
import {
    archiveFramesError,
    archiveFramesSuccess,
    BIKE_DELETE_REQUESTED, BIKE_REVIEW_BIKE, BIKE_REVIEW_REQUESTED,
    deleteBikesError,
    deleteBikesSuccess,
    deleteFramesError,
    deleteFramesSuccess,
    FRAME_ARCHIVE_REQUESTED,
    FRAME_DELETE_REQUESTED,
    FRAME_LIST_REQUESTED,
    FRAME_SAVE_REQUESTED,
    FRAME_UPLOAD_REQUESTED,
    getFrameListError,
    getFrameListOK, reviewBikeError, reviewBikeOK,
    saveFrameError,
    saveFrameSuccess,
    uploadFrameError,
    uploadFrameSuccess,
    reviewBike,
} from "../actions/bike";

export function* reviewBikeStart(action) {
    const bikeIdsToReview = action.payload.bikeReviewList;
    if (bikeIdsToReview && bikeIdsToReview.length > 0) {
        yield put(reviewBike(bikeIdsToReview[0]));
        yield call(history.push, "/bike-browse");
    } else{
        yield put(reviewBikeError("No Bikes to Review"));
    }

}

export function* watchForReviewBikeStart() {
    yield takeLatest(BIKE_REVIEW_REQUESTED, reviewBikeStart);
}

export function* reviewBikeParts(action) {
    try {
        const token = yield select(selectors.token);
        if (token) {
            const completePayload = Object.assign(action.payload, { token });
            const response = yield call(bike.getBikeParts, completePayload);
            yield put(reviewBikeOK(response.data));
        } else {
            yield call(history.push, "/login");
        }
    } catch (error) {
        yield put(reviewBikeError("Get Bike for Review failed"));
    }
}

export function* watchForReviewBike() {
    yield takeLatest(BIKE_REVIEW_BIKE, reviewBikeParts);
}

export function* deleteBikes(bikeIdsToDelete, token) {
    try {
        for (let i = 0; i < bikeIdsToDelete.length; i++) {
            yield call(bike.deleteBike, { bikeId: bikeIdsToDelete[i], token });
        }

    } catch (error) {
        yield put(deleteBikesError("Delete Frames failed"));
    }
}

export function* deleteBikesAndGetList(action) {
    try {
        const token = yield select(selectors.token);
        if (token) {
            const bikeIdsToDelete = action.payload.bikeDeleteList;
            const searchCriteria = action.payload.searchCriteria;

            if (bikeIdsToDelete && bikeIdsToDelete.length > 0) {
                yield* deleteBikes(bikeIdsToDelete, token);
            }

            if (searchCriteria && searchCriteria.brand) {
                const searchPayload = Object.assign(searchCriteria, { token });
                const searchResponse = yield call(bike.getFrames, searchPayload);
                yield put(getFrameListOK(searchResponse.data));
            } else {
                yield put(deleteBikesSuccess());
            }
        } else {
            yield call(history.push, "/login");
        }
    } catch (error) {
        yield put(deleteBikesError("Delete Bikes failed"));
    }
}

export function* watchForDeleteBikes() {
    yield takeLatest(BIKE_DELETE_REQUESTED, deleteBikesAndGetList);
}

export function* deleteFrames(frameIdsToDelete, token) {
    try {
        for (let i = 0; i < frameIdsToDelete.length; i++) {
            yield call(bike.deleteFrame, { frameId: frameIdsToDelete[i], token });
        }

    } catch (error) {
        yield put(deleteFramesError("Delete Frames failed"));
    }
}

export function* deleteFramesAndGetList(action) {
    try {
        const token = yield select(selectors.token);
        if (token) {
            const frameIdsToDelete = action.payload.frameDeleteList;
            const searchCriteria = action.payload.searchCriteria;

            if (frameIdsToDelete && frameIdsToDelete.length > 0) {
                yield* deleteFrames(frameIdsToDelete, token);
            }

            if (searchCriteria && searchCriteria.brand) {
                const searchPayload = Object.assign(searchCriteria, { token });
                const searchResponse = yield call(bike.getFrames, searchPayload);
                yield put(getFrameListOK(searchResponse.data));
            } else {
                yield put(deleteFramesSuccess());
            }
        } else {
            yield call(history.push, "/login");
        }
    } catch (error) {
        yield put(deleteFramesError("Delete Frames failed"));
    }
}

export function* watchForDeleteFrames() {
    yield takeLatest(FRAME_DELETE_REQUESTED, deleteFramesAndGetList);
}
export function* archiveFrames(frameIdsToArchive, token) {
    try {
        for (let i = 0; i < frameIdsToArchive.length; i++) {
            const frame = { id: frameIdsToArchive[i], archived: true };
            yield call(bike.saveFrame, { frame, token });
        }

    } catch (error) {
        yield put(archiveFramesError("Archive Frames failed"));
    }
}

export function* archiveFramesAndGetList(action) {
    try {
        const token = yield select(selectors.token);
        if (token) {
            const frameIdsToArchive = action.payload.frameArchiveList;
            const searchCriteria = action.payload.searchCriteria;

            if (frameIdsToArchive && frameIdsToArchive.length > 0) {
                yield* archiveFrames(frameIdsToArchive, token);
            }

            if (searchCriteria && searchCriteria.brand) {
                const searchPayload = Object.assign(searchCriteria, { token });
                const searchResponse = yield call(bike.getFrames, searchPayload);
                yield put(getFrameListOK(searchResponse.data));
            } else {
                yield put(archiveFramesSuccess());
            }
        } else {
            yield call(history.push, "/login");
        }
    } catch (error) {
        yield put(archiveFramesError("Archive Frames failed"));
    }
}

export function* watchForArchiveFrames() {
    yield takeLatest(FRAME_ARCHIVE_REQUESTED, archiveFramesAndGetList);
}

export function* saveFrame(action) {
    try {
        const token = yield select(selectors.token);
        if (token) {
            const completePayload = Object.assign(action.payload, { token });
            const response = yield call(bike.saveFrame, completePayload);
            if (completePayload.searchCriteria) {
                const searchPayload = Object.assign(completePayload.searchCriteria, { token });
                const searchResponse = yield call(bike.getFrames, searchPayload);
                yield put(getFrameListOK(searchResponse.data));
            } else {
                yield put(saveFrameSuccess(response.data));
            }
        } else {
            yield call(history.push, "/login");
        }
    } catch (error) {
        yield put(saveFrameError("Save Frame failed"));
    }
}

export function* watchForSaveFrame() {
    yield takeLatest(FRAME_SAVE_REQUESTED, saveFrame);
}

export function* uploadFrame(action) {
    try {
        const token = yield select(selectors.token);
        if (token) {
            const completePayload = Object.assign(action.payload, { token });
            const response = yield call(bike.uploadFrame, completePayload);
            yield put(uploadFrameSuccess(response.data));
        } else {
            yield call(history.push, "/login");
        }
    } catch (error) {
        yield put(uploadFrameError("Save Frame failed"));
    }
}

export function* watchForUploadFrame() {
    yield takeLatest(FRAME_UPLOAD_REQUESTED, uploadFrame);
}

export function* getFrames(action) {
    try {
        const token = yield select(selectors.token);
        if (token) {
            const completePayload = Object.assign(action.payload, { token });
            const response = yield call(bike.getFrames, completePayload);
            yield put(getFrameListOK(response.data));
        } else {
            yield call(history.push, "/login");
        }
    } catch (error) {
        yield put(getFrameListError("Get Frames failed"));
    }
}

export function* watchForGetFrames() {
    yield takeLatest(FRAME_LIST_REQUESTED, getFrames);
}
