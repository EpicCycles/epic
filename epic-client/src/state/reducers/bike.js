import {CLEAR_ALL_STATE} from "../actions/application";
import {
    CLEAR_FRAME, FRAME_ARCHIVE_ERROR,
    FRAME_LIST_ERROR,
    FRAME_LIST_OK,
    FRAME_LIST_REQUESTED,
    FRAME_SAVE_ERROR,
    FRAME_SAVE_OK,
    FRAME_SAVE_REQUESTED, FRAME_UPLOAD_ERROR, FRAME_UPLOAD_OK, FRAME_UPLOAD_REQUESTED
} from "../actions/bike";
import {removeKey} from "../../helpers/utils";

const initialState = {
    isLoading: false
};

const bike = (state = initialState, action) => {
    switch (action.type) {
        case CLEAR_ALL_STATE:
        case CLEAR_FRAME:
            return initialState;
        case FRAME_SAVE_REQUESTED:
        case FRAME_UPLOAD_REQUESTED:
            return {
                ...state,
                frame: action.payload.frame,
                isLoading: true,
            };
        case FRAME_SAVE_OK:
        case FRAME_UPLOAD_OK:
            return {
                ...state,
                frame: action.payload,
                isLoading: false,
            };
        case FRAME_LIST_REQUESTED:
            return {
                ...state,
                isLoading: true,
            };
        case FRAME_LIST_OK:
            return {
                ...state,
                frames: action.payload,
                isLoading: false,
            };
        case FRAME_SAVE_ERROR:
        case FRAME_UPLOAD_ERROR:
        case FRAME_LIST_ERROR:
        case FRAME_ARCHIVE_ERROR:
            return {
                ...state,
                isLoading: false,
            };
        default:
            return state;
    }
};

export default bike;