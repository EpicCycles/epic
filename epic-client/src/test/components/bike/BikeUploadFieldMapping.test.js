import React from 'react';
import {COLOURS_FIELD} from "../../../helpers/models";
import {BikeUploadFieldMapping} from "../../../components/bike/BikeUploadFieldMapping";

test("it renders correctly when no rows match", () => {
    const field = COLOURS_FIELD;
    const rowMappings = [];
    const component = shallow(<BikeUploadFieldMapping
        field={field}
        rowMappings={rowMappings}
    />);
    expect(component).toMatchSnapshot();
});
test("it renders correctly when rows match", () => {
    const field = COLOURS_FIELD;
    const rowMappings = [{fieldName:"blah"}];
    const component = shallow(<BikeUploadFieldMapping
        field={field}
        rowMappings={rowMappings}
    />);
    expect(component).toMatchSnapshot();
});