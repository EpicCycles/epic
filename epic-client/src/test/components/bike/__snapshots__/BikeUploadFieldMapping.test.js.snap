// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly when no rows match 1`] = `
<div
  className="grid-row"
>
  <div
    className="grid-item--borderless field-label align_right"
    key="bikeFieldundefined"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    Colours
  </div>
  <div
    className="grid-item--borderless field-label "
    key="matchFieldDivundefined"
  />
</div>
`;

exports[`it renders correctly when rows match 1`] = `
<div
  className="grid-row"
>
  <div
    className="grid-item--borderless field-label align_right"
    key="bikeFieldundefined"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    Colours
  </div>
  <div
    className="grid-item--borderless field-label "
    key="matchFieldDivundefined"
  >
    <BikeUploadMatchedData
      key="matchAreaundefined0"
      matchIndex={0}
      matched={
        Object {
          "fieldName": "blah",
        }
      }
    />
  </div>
</div>
`;
